app/controllers/v0/claim_documents_controller.rb

```diff
@@ -2,36 +2,45 @@
 
 require 'pension_burial/tag_sentry'
 require 'lgy/tag_sentry'
+require 'claim_documents/monitor'
+require 'lighthouse/benefits_intake/service'
+require 'pdf_utilities/datestamp_pdf'
 
 module V0
   class ClaimDocumentsController < ApplicationController
     service_tag 'claims-shared'
     skip_before_action(:authenticate)
+    before_action :load_user
 
     def create
-      Rails.logger.info "Creating PersistentAttachment FormID=#{form_id}"
+      uploads_monitor.track_document_upload_attempt(form_id, current_user)
 
-      attachment = klass.new(form_id:)
+      @attachment = klass&.new(form_id:)
       # add the file after so that we have a form_id and guid for the uploader to use
-      attachment.file = unlock_file(params['file'], params['password'])
+      @attachment.file = unlock_file(params['file'], params['password'])
 
-      raise Common::Exceptions::ValidationErrors, attachment unless attachment.valid?
+      if Flipper.enabled?(:document_upload_validation_enabled) && !stamped_pdf_valid? &&
+         %w[21P-527EZ 21P-530 21P-530V2].include?(form_id)
```

Comment:
the form_id check should come first in the conditional list, so if that fails the later ones will not be checked.  will prevent an unneeded `stamped_pdf_valid?` call